src/test/foldingProvider.test.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert'
+import 'mocha'
+import * as vscode from 'vscode'
+import AsciidocFoldingProvider from '../features/foldingProvider'
+import { createNewAsciidocEngine } from './engine'
+import { InMemoryDocument } from './inMemoryDocument'
+
+const testFileName = vscode.Uri.file('test.adoc')
+
+suite('asciidoc.FoldingProvider', () => {
+  test('Should not return anything for empty document', async () => {
+    const folds = await getFoldsForDocument('')
+    assert.strictEqual(folds.length, 0)
+  })
+
+  test('Should not return anything for document without headers', async () => {
+    const folds = await getFoldsForDocument(`a
+*b* afas
+a=b
+a`)
+    assert.strictEqual(folds.length, 0)
+  })
+
+  test('Should fold from header to end of document', async () => {
+    const folds = await getFoldsForDocument(`= a
+
+== b
+
+c
+d`)
+    assert.strictEqual(folds.length, 2)
+    const firstFold = folds[0]
+    assert.strictEqual(firstFold.start, 0)
+    assert.strictEqual(firstFold.end, 5)
+  })
+
+  test('Should leave single newline before next header', async () => {
+    const folds = await getFoldsForDocument(`
+== a
+x
+
+== b
+y`)
+    assert.strictEqual(folds.length, 2)
+    const firstFold = folds[0]
+    assert.strictEqual(firstFold.start, 1)
+    assert.strictEqual(firstFold.end, 3)
+  })
+
+  test('Should collapse multiple newlines to single newline before next header', async () => {
+    const folds = await getFoldsForDocument(`
+= a
+x
+
+
+
+= b
+y`)
```

Comment:
It should be:

```suggestion
== a
x



== b
y`)
```

We cannot have more than one document title in AsciiDoc.